Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -157,6 +332,19 @@ namespace GradientSignal
         GradientRequestBus::Handler::BusConnect(GetEntityId());
 
         UpdatePreviewSettings();
+
+        // If we have a valid output image path set and the image doesn't exist,
+        // then bake it when we activate our component.
+        if (!m_configuration.m_outputImagePath.empty())
+        {
+            AZ::IO::Path fullPathIO = AzToolsFramework::GetAbsolutePathFromRelativePath(m_configuration.m_outputImagePath);
+            if (!AZ::IO::SystemFile::Exists(fullPathIO.c_str()))
+            {
+                QTimer::singleShot(0, [this]() {
+                    BakeImage();
+                });
```

Comment:
This seems potentially a bit dangerous if the component gets deactivated or destroyed before the singleShot executes. It might be safer to connect the component to the TickBus (similar to what BakeImage() itself does), and then just execute the bake on the next tick. The difference is that component deactivation will cause it to disconnect from the TickBus, so the bake would never execute.

Reply:
Good call-out, I can re-work this